src/components/forms/EventStreamsForm.js

```diff
@@ -0,0 +1,185 @@
+import React, { Component } from "react";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import { IcseNameInput } from "../Inputs";
+import { EndpointSelect, IcseSelect } from "../Dropdowns";
+import { IcseFormGroup } from "../Utils";
+import { ToolTipWrapper } from "../Tooltips";
+import { isIpStringInvalid } from "../../lib/iam-utils";
+import { TextArea } from "@carbon/react";
+import { titleCase } from "lazy-z";
+import PropTypes from "prop-types";
+
+/**
+ * EventStreamsForm
+ */
+class EventStreamsForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = this.props.data;
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleAllowedIps = this.handleAllowedIps.bind(this);
+    this.handlePlanChange = this.handlePlanChange.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  /**
+   * Handle input change for allowed ips text field
+   * @param {event} event
+   */
+  handleAllowedIps(event) {
+    // removing white space and checking for empty value
+    let value = event.target.value.replace(/\s*/g, "");
+    if (value === "") value = null;
+    this.setState({ private_ip_allowlist: value });
+  }
+
+  /**
+   * handle input change
+   * @param {event} event event
+   */
+  handleInputChange(event) {
+    this.setState(this.eventTargetToNameAndValue(event));
+  }
+
+  /**
+   * Handle input change for a select
+   * @param {event} event
+   */
+  handlePlanChange(event) {
+    let item = event.target.value.toLowerCase();
+    this.setState({ plan: item });
+    if (item !== "enterprise") {
+      this.setState({
+        throughput: "",
+        storage_size: "",
+        endpoint: "",
+        private_ip_allowlist: "",
+      });
+    }
+  }
+
```

Comment:
can we make an object that contains everything that might be set into state and only add the throughput, storage_size, endpoint, and private_ip_allowlist iff item === enterprise? to avoid setting state twice?

Reply:
How does this look?
```js
  handlePlanChange(event) {
    let tempState = { plan: event.target.value.toLowerCase() };
    if (item !== "enterprise") {
      tempState = {
        ...tempState,
        throughput: "",
        storage_size: "",
        endpoint: "",
        private_ip_allowlist: "",
      };
    }
    this.setState(tempState);
  }
```